buildSrc/src/main/groovy/org/elasticsearch/gradle/BuildPlugin.groovy

```diff
@@ -1102,4 +1070,33 @@ class BuildPlugin implements Plugin<Project> {
             deps.mappings = project.dependencyLicenses.mappings
         }
     }
+
+    /**
+     * Pins the test seed at configuration time so it isn't different on every
+     * {@link Test} execution. This is useful if random
+     * decisions in one run of {@linkplain Test} influence the
+     * outcome of subsequent runs. Pinning the seed up front like this makes
+     * the reproduction line from one run be useful on another run.
+     */
+    static String setupSeed(Project project) {
+        if (project.rootProject.ext.has('testSeed')) {
+            /* Skip this if we've already pinned the testSeed. It is important
+             * that this checks the rootProject so that we know we've only ever
+             * initialized one time. */
+            return project.rootProject.ext.testSeed
+        }
+        String testSeed = System.getProperty('tests.seed')
+        if (testSeed == null) {
+            long seed = new Random(System.currentTimeMillis()).nextLong()
+            testSeed = Long.toUnsignedString(seed, 16).toUpperCase(Locale.ROOT)
+        }
+        /* Set the testSeed on the root project first so other projects can use
+         * it during initialization. */
+        project.rootProject.ext.testSeed = testSeed
+        project.rootProject.subprojects {
```

Comment:
Why set it on each project if we only look for it on the root ?

Reply:
This was historical. I agree, I don't think it's necessary.